### ISSUE TITLE
Exclude does not work with module substitution

### ISSUE THREAD

**Issue by jub77** [spring-gradle-plugins/dependency-management-plugin#190]
When using dependency with project (in multiproject build) - the exclude feature is correctly used. But when used with module substitution, the exclude definition is ignored.

See attached
[exclude_problem.zip](https://github.com/spring-gradle-plugins/dependency-management-plugin/files/1404901/exclude_problem.zip)

The project in project_dependency executing `gradle :b:dependecies` produces
```
\--- project :a
     \--- com.google.guava:guava: -> 23.2-jre
```

But when in project artifact_dependency_substitute executing `gradle :b:dependencies` I get
```
\--- dep.test:a: -> project :a
     \--- com.google.guava:guava: -> 23.2-jre
          +--- com.google.code.findbugs:jsr305:1.3.9
          +--- com.google.errorprone:error_prone_annotations:2.0.18
          +--- com.google.j2objc:j2objc-annotations:1.1
          \--- org.codehaus.mojo:animal-sniffer-annotations:1.14
```
The exclude definition is ignored.

Tested in on Gradle 4.2 with plugin version 1.0.3

**Comment by wilkinsona** [comment]
Thanks for the sample. 

#### TL;DR

This is working as designed and you can avoid the problem with a little more configuration. You need to configure the resolution strategy that the dependency management plugin uses for any configurations that it creates and resolves with the same module substitution:

```groovy
dependencyManagement {
	resolutionStrategy {
		dependencySubstitution {
			substitute module('dep.test:a') with project(':a')
		}
	}
}
```

#### Longer explanation

When exclusions are being applied, the dependency management plugin makes a copy of the configuration that's being resolved and resolves the copy. The use of a copy is necessary so that the configuration's dependencies can be introspected and the exclusions correctly applied to the original configuration before it's resolved. The configuration is intentionally copied in such a way that the resolution strategy is not copied over. This provides complete, independent control of the resolution strategy and avoids the problem reported in #61. Unfortunately, this flexibility does come at the cost of the dependency substitution having to be configured in two places.

**Comment by jub77** [comment]
Thank you for the explanation.
